rasa/core/channels/console.py

```diff
@@ -142,14 +146,21 @@ async def _send_message_receive_stream(
                     yield json.loads(line.decode(DEFAULT_ENCODING))
 
 
-def _get_stream_reading_timeout() -> ClientTimeout:
-    timeout_in_seconds = int(
-        os.environ.get(
-            STREAM_READING_TIMEOUT_ENV, DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS
```

Comment:
I hadn't realised there was a `STREAM_READING_TIMEOUT_ENV` env variable which can be used to set console channel timeout 😳 

The problem is I think all these settings such as STREAM_READING_TIMEOUT_ENV, DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS, and DEFAULT_REQUEST_TIMEOUT are confusing; semantically they mean the same thing though in practice they are not used consistently. What I mean is, stream reading timeout sounds like an even more general setting than request timeout but it is only used for console channel, not for example for action server calls. It's a little confusing as one is on the client side (shell) and the other on the server (Rasa calling action server) but both are controlled by the same CLI and setting semantically applies to requests made from both client and server. 

@joejuzl may have better ideas around this.

Reply:
I think possibly changing them to make them more semantically meaningful might be a good idea, but since it would be a breaking change we'd probably have to plan it out a bit more carefully than can be done in the scope of this ticket